modules/gis/src/lib/geojson-to-flat-geojson.ts

```diff
@@ -0,0 +1,122 @@
+import {getPolygonSignedArea} from '@math.gl/polygon';
+
+import {Feature, Position, FlatFeature} from '@loaders.gl/schema';
+
+export type GeojsonToFlatGeojsonOptions = {
+  coordLength: number;
+  fixRingWinding: boolean;
+};
+
+export function geojsonToFlatGeojson(
+  features: Feature[],
+  options: GeojsonToFlatGeojsonOptions = {coordLength: 2, fixRingWinding: true}
+): FlatFeature[] {
+  return features.map((feature) => flattenFeature(feature, options));
+}
+
+function flattenPoint(
+  coordinates: Position,
+  data: number[],
+  indices: number[],
+  options: GeojsonToFlatGeojsonOptions
+) {
+  indices.push(data.length);
+  data.push(...coordinates);
+
+  // Pad up to coordLength
+  for (let i = coordinates.length; i < options.coordLength; i++) {
+    data.push(0);
+  }
+}
+
+function flattenLineString(
+  coordinates: Position[],
+  data: number[],
+  indices: number[],
+  options: GeojsonToFlatGeojsonOptions
+) {
+  indices.push(data.length);
+  for (const c of coordinates) {
+    data.push(...c);
+
+    // Pad up to coordLength
+    for (let i = c.length; i < options.coordLength; i++) {
+      data.push(0);
+    }
+  }
+}
+
+function flattenPolygon(
+  coordinates: Position[][],
```

Comment:
I'd create internal types for this flatten functions

Reply:
Done